src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/ReleaseContentBlockService.cs

```diff
@@ -0,0 +1,213 @@
+                releaseId: block.ContentSection!.Release!.ReleaseId,
+                locked: now,
+                lockedUntil: block.LockedUntil!.Value,
+                lockedBy: new UserDetailsViewModel(user)
+            );
+
+            await _hubContext.Clients
+                .Group(viewModel.ReleaseId.ToString())
+                .ContentBlockLocked(viewModel);
+
+            return viewModel;
+        }
+
+        /// <summary>
+        /// Try to unlock a content block, notifying any subscribed hub
+        /// clients that the lock has been acquired.
+        /// </summary>
+        /// <returns>Nothing if the content block was successfully unlocked.</returns>
+        public async Task<Either<ActionResult, Unit>> UnlockContentBlock(Guid id, bool force = false)
+        {
+            var userId = _userService.GetUserId();
+
+            return await GetContentBlock(id)
+                .OnSuccessDo(CheckCanUpdateBlock)
+                .OnSuccessDo(block => CheckNoExistingLock(block, userId, force))
+                .OnSuccessVoid(
+                    async block =>
+                    {
+                        _contentDbContext.ContentBlocks.Update(block);
+
+                        block.Locked = null;
+                        block.LockedById = null;
+
+                        await _contentDbContext.SaveChangesAsync();
+
+                        var viewModel = new ReleaseContentBlockUnlockViewModel(
+                            id: block.Id,
+                            sectionId: block.ContentSection!.Id,
+                            releaseId: block.ContentSection!.Release!.ReleaseId
+                        );
+
+                        await _hubContext.Clients
+                            .Group(viewModel.ReleaseId.ToString())
+                            .ContentBlockUnlocked(viewModel);
+                    }
+                );
+        }
+
+        private async Task<Either<ActionResult, ContentBlock>> GetContentBlock(Guid contentBlockId)
+        {
+            return await _persistenceHelper.CheckEntityExists<ContentBlock>(
+                contentBlockId,
+                q =>
+                    q.Include(block => block.ContentSection)
+                        .ThenInclude(contentSection => contentSection!.Release)
+                        .ThenInclude(releaseContentSection => releaseContentSection.Release)
+                        .Include(block => block.LockedBy)
+            );
+        }
+
+        private async Task<Either<ActionResult, Unit>> CheckCanUpdateBlock(ContentBlock contentBlock)
+        {
+            if (contentBlock.ContentSection?.Release.Release is null)
+            {
+                return new ForbidResult();
+            }
+
+            return await _userService
+                .CheckCanUpdateRelease(contentBlock.ContentSection.Release.Release)
+                .OnSuccessVoid();
+        }
+
+        private bool TryFindExistingLock(
```

Comment:
Yeh good point, it is a little confusing, probably because the code changed a few times before it settled onto its current form!

I've renamed this to `TryFindConflictingLock` if that sounds suitable? This would typically mean a lock owned by another user.

Reply:
Yeah that sounds exactly right, ta.